guava/src/com/google/common/util/concurrent/Uninterruptibles.java

```diff
@@ -326,7 +326,67 @@ public static boolean tryAcquireUninterruptibly(
     }
   }
 
-  // TODO(user): Add support for waitUninterruptibly.
+  /**
+   * Invokes {@code object.}{@link Object#wait() wait()} uninterruptibly.
+   *
+   * @param object to wait upon.
+   *
+   * @throws IllegalMonitorStateException if the current thread is not
+   *               the owner of the object's monitor.
+   */
+  @GwtIncompatible // concurrency
+  public static void waitUninterruptibly(Object object) {
+    Preconditions.checkNotNull(object);
+    boolean interrupted = false;
+    try {
+      while (true) {
+        try {
+          object.wait();
+          return;
+        } catch (InterruptedException e) {
+          interrupted = true;
```

Comment:
The general opinion internally seems to be that, if people are willing to change their code, they should probably move to something like `Monitor`. Arguably we shouldn't even provide `waitUninterruptibly(Object)`, but I can at least see the value in providing a method that users can easily plug in for a broken `try-catch` block without otherwise touching the structure of their code and without introducing a usage of `Supplier` (which we'd like to phase out after Java 8).